InnerEye/ML/utils/supervised_criterion.py

```diff
@@ -56,10 +56,28 @@ def forward_minibatch(self, output: Any, target: Any, **kwargs: Any) -> Any:
 class BinaryCrossEntropyWithLogitsLoss(SupervisedLearningCriterion):
     """A wrapper function for torch.nn.BCEWithLogitsLoss to enable label smoothing"""
 
-    def __init__(self, class_counts: Optional[Dict[float, float]] = None, **kwargs: Any):
+    def __init__(self, num_classes: int,
+                 class_counts: Optional[Dict[float, int]] = None,
+                 num_train_samples: Optional[int] = None,
+                 **kwargs: Any):
+        """
+        :param num_classes: The number of classes the model predicts. For binary classification num_classes is one
+                            and for multi-label classification tasks num_classes will be greater than one.
+                            Note that in multi-label tasks, the negative class (all other ground truth classes being
```

Comment:
not sure i get what "the negative class should not be a separate class means"

Reply:
I've moved this bit to `creating_dataset.md`, it makes more sense there.